service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapper.java

```diff
@@ -128,22 +128,35 @@ private String mapTopicListToComponent(ListResource topicList) {
     }
 
     private String mapTopicListComponents(ListResource topicList) {
-        return mapCategorizedResources(topicList) + mapListResourceToComponents(topicList);
+        return String.join(
+            StringUtils.EMPTY,
+            mapCategorizedResources(topicList),
+            mapContainedResources(topicList),
+            mapUncategorizedResources(topicList)
+        );
     }
 
     private String mapCategorizedResources(ListResource topicList) {
         return topicList.getEntry().stream()
             .map(EncounterComponentsMapper::getReferenceElement)
-            .filter(reference -> reference.getValue().matches(LIST_REFERENCE_PATTERN))
+            .filter(reference -> reference
+                .getValue()
+                .matches(LIST_REFERENCE_PATTERN)
+            )
             .map(this::getRequiredResource)
-            .filter(resource -> ResourceType.List.equals(resource.getResourceType()))
             .map(ListResource.class::cast)
-            .filter(listResource -> CodeableConceptMappingUtils.hasCode(listResource.getCode(), List.of(CATEGORY_LIST_CODE)))
             .map(this::mapCategoryListToComponent)
             .collect(Collectors.joining());
     }
 
     private String mapCategoryListToComponent(ListResource categoryList) {
+        if (!CodeableConceptMappingUtils.hasCode(categoryList.getCode(), List.of(CATEGORY_LIST_CODE))) {
+            throw new EhrMapperException(
+                    "Unexpected list %s referenced in Consultation, expected list to be coded as Category (EHR) or be a container"
```

Comment:
I think the exception could be improved. The container part seems strange to me as we already know it's a list. Also being clear that we found this resource as an entry within a Topic list, might help clarify what the situation was.

From:
"Unexpected list NaughtyList referenced in Consultation, expected list to be coded as Category (EHR) or be a container"

To:
"List resource with id "NaughtyList" as entry within List (Topic) is required to be coded as List (Category) 24781000000107 to match GP Connect Specification, but no such code found on resource."